community/cypher/cypher/src/test/scala/org/neo4j/cypher/EagerizationAcceptanceTest.scala

```diff
@@ -2323,6 +2323,39 @@ class EagerizationAcceptanceTest
     assertNumberOfEagerness(query, 2, optimalEagerCount = 1)
   }
 
+  test("should always be eager after deleted relationships if there are any subsequent expands that might load them") {
+    val device = createLabeledNode("Device")
+    val cookies = (0 until 2).foldLeft(Map.empty[String, Node]) { (nodes, index) =>
+      val name = s"c$index"
+      val cookie = createLabeledNode(Map("name" -> name), "Cookie")
+      relate(device, cookie)
+      relate(cookie, createNode())
+      nodes + (name -> cookie)
+    }
+
+    val query =
+      """
+        |MATCH (c:Cookie {name: {cookie}})<-[r2]-(d:Device)
```

Comment:
It helps clarify things when debugging, since the difference between working and failing queries was based on whether r and r2 were in the same or different rows.